### PR TITLE
SW-5672 Reset species list deliverable submission when new species are added

### PR DESCRIPTION
**Pull request by nickgraz**
If a species is added to a participant project, and there is an associated deliverable ID (active or recent), and there is a submission that is "Approved" for the deliverable / project, "reset" the submission to the "Not Submitted" status.

### CODE REVIEW

**Comment by nickgraz** [comment]
* **#2229** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2229?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2228** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2228?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @nickgraz and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/main/kotlin/com/terraformation/backend/accelerator/SpeciesSubmissionUpdater.kt [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+    val (feedback, internalComment, submissionStatus) =
+        with(SUBMISSIONS) {
+          dslContext
+              .select(FEEDBACK, INTERNAL_COMMENT, SUBMISSION_STATUS_ID)
+              .from(SUBMISSIONS)
+              .where(PROJECT_ID.eq(projectId))
+              .and(DELIVERABLE_ID.eq(deliverableId))
+              .fetchOne()
+              ?.map { Triple(it[FEEDBACK], it[INTERNAL_COMMENT], it[SUBMISSION_STATUS_ID]!!) }
```
Comment (sgrimm) [inline src/main/kotlin/com/terraformation/backend/accelerator/SpeciesSubmissionUpdater.kt]: I don't think you need this `map` call -- jOOQ's `Record` classes support Kotlin destructuring.

File: src/test/kotlin/com/terraformation/backend/accelerator/SpeciesSubmissionUpdaterTest.kt [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+  fun setUp() {
+    insertOrganization()
+
+    every { user.canReadProject(any()) } returns true
+    every { user.canReadSubmission(any()) } returns true
+    every { user.canUpdateSubmissionStatus(any(), any()) } returns true
+  }
+
+  @Nested
+  inner class NotifyIfNoNewerAddedSpecies {
```
Comment (sgrimm) [inline src/test/kotlin/com/terraformation/backend/accelerator/SpeciesSubmissionUpdaterTest.kt]: This class name seems inaccurate.

File: src/test/kotlin/com/terraformation/backend/accelerator/SpeciesSubmissionUpdaterTest.kt [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+              deliverableId = deliverableId,
+              feedback = "So far so good",
+              id = submissionId,
+              internalComment = "Internal comment",
+              modifiedBy = userId,
+              modifiedTime = now,
+              projectId = projectId,
+              submissionStatusId = SubmissionStatus.InReview)
+
+      assertEquals(expected, actual, "The submission's status was not updated to 'Not Submitted'")
```
Comment (sgrimm) [inline src/test/kotlin/com/terraformation/backend/accelerator/SpeciesSubmissionUpdaterTest.kt]: The assertion message here is the same as on the other test case but doesn't seem correct in this context.

**Review by sgrimm — CHANGES_REQUESTED** [review]
Seems straightforward enough. Just a few small comments.

File: src/test/kotlin/com/terraformation/backend/accelerator/SpeciesSubmissionUpdaterTest.kt (hunk 3 above)
Comment (nickgraz) [reply src/test/kotlin/com/terraformation/backend/accelerator/SpeciesSubmissionUpdaterTest.kt]: I simplified the test, hopefully it makes more sense now

**Review by sgrimm — APPROVED** [review]

**Comment by nickgraz** [comment]
### Merge activity

* **Aug 2, 5:49 PM EDT**: @nickgraz started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/2228).
* **Aug 2, 5:50 PM EDT**: @nickgraz merged this pull request with [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/2228).

**Pull request merged**